services/src/main/java/org/keycloak/services/HttpRequestImpl.java

```diff
@@ -59,6 +59,10 @@ public MultivaluedMap<String, String> getDecodedFormParameters() {
         if (delegate == null) {
             return null;
         }
+        MediaType mediaType = getHttpHeaders().getMediaType();
+        if (mediaType.isCompatible(MediaType.valueOf("application/x-www-form-urlencoded"))) {
```

Comment:
Couldn't mediaType be null?

Reply:
@sschu you're right you have to check even if it's not null, I added the change